app/scripts/metamask-controller.js

```diff
@@ -4885,52 +4892,76 @@ export default class MetamaskController extends EventEmitter {
   // Hardware
   //
 
-  async getKeyringForDevice(deviceName, hdPath = null) {
+  /**
+   * Select a hardware wallet device and execute a
+   * callback with the keyring for that device.
+   *
+   * Note that KeyringController state is not updated before
+   * the end of the callback execution, and calls to KeyringController
+   * methods within the callback can lead to deadlocks.
+   *
+   * @param {object} options - The options for the device
+   * @param {string} options.name - The device name to select
+   * @param {string} options.hdPath - An optional hd path to be set on the device
+   * keyring
+   * @param {*} callback - The callback to execute with the keyring
+   * @returns {*} The result of the callback
+   */
+  async withKeyringForDevice(options, callback) {
```

Comment:
This is probably the main change of this PR: `getKeyringForDevice` is now `withKeyringForDevice`.

Instead of being a getter for a Keyring instance related to a device, it is now a safe callback executor for hardware devices

Reply:
Note that this has been then renamed to `#withKeyringForDevice` in this commit: https://github.com/MetaMask/metamask-extension/commit/bfda15f6caffa7648eb2ff32a735c8df8fc4f500